module/users/helper.js

```diff
@@ -570,58 +584,62 @@ module.exports = class UsersHelper {
    * @param {String} role - role.
    * @returns {Object} returns a list of entity type by location and role.
    */
-
-  static entityTypesByLocationAndRole(stateLocationId, role) {
+   static entityTypesByLocationAndRole(stateLocationId, role) {
     return new Promise(async (resolve, reject) => {
       try {
-        let filterQuery = {
-          "registryDetails.code": stateLocationId,
-        };
-
-        if (gen.utils.checkValidUUID(stateLocationId)) {
-          filterQuery = {
-            "registryDetails.locationId": stateLocationId,
-          };
-        }
-
-        const entitiesData = await entitiesHelper.entityDocuments(filterQuery, [
-          "_id",
-        ]);
-
-        if (!entitiesData.length > 0) {
-          throw {
-            message: constants.apiResponses.ENTITIES_NOT_EXIST_IN_LOCATION,
-          };
-        }
-
+        let entityKey = constants.common.SUBENTITY + stateLocationId;
         const rolesDocument = await userRolesHelper.roleDocuments(
           {
             code: role.toUpperCase(),
           },
           ["_id", "entityTypes.entityType"]
         );
-
+        
         if (!rolesDocument.length > 0) {
           throw {
             message: constants.apiResponses.USER_ROLES_NOT_FOUND
           };
         }
+        
 
+
+        let bodyData={};    
+        if (gen.utils.checkValidUUID(stateLocationId)) {
```

Comment:
@VISHNUDAS-tunerlabs this condition needs to be change according to type school and call orgs service

Reply:
need to check always passing stateLocationId - with app team